src/datasource.ts

```diff
@@ -66,6 +66,9 @@ export class DataFrameDataSource extends DataSourceApi<DataframeQuery> {
 
   async getTableMetadata(id: string) {
     const resolvedId = getTemplateSrv().replace(id);
+    if (!resolvedId) {
```

Comment:
If the query's ID is a variable with no value, this was calling the /tables route pointlessly